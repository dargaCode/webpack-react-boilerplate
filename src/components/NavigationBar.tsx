import React from "react";
import { NavLink } from "react-router-dom";
import "./_general.scss";
import "./_NavigationBar.scss";

export default function NavigationBar(): JSX.Element {
  return (
    <header>
      <nav>
        <ul>
          <li>
            <NavLink to="/home" activeClassName="activeNavSection">
              home
            </NavLink>
          </li>
          <li>
            <NavLink to="/about" activeClassName="activeNavSection">
              about
            </NavLink>
          </li>
          <li>
            <NavLink to="/other" activeClassName="activeNavSection">
              other?
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>
  );
}
