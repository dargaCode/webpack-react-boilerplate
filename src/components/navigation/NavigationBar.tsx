import React from "react";
import { NavLink } from "react-router-dom";
import "../../config/_general.scss";
import styles from "./NavigationBar.module.scss";

export default function NavigationBar(): JSX.Element {
  return (
    <nav className={styles.bar}>
      <div className="nocontent">
        {/* ignored by google */}
        <h2 className={styles.invisibleButOutlineReadable}>Navigation</h2>
      </div>
      <div className={styles.container}>
        <ul>
          <li className={styles.button}>
            <NavLink to="/hello" activeClassName="activeNavSection">
              Hello
            </NavLink>
          </li>

          <li className={styles.button}>
            <NavLink to="/world" activeClassName="activeNavSection">
              World
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
}
