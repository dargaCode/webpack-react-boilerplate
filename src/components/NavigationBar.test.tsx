import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import NavigationBar from "./NavigationBar";

describe("<NavigationBar/>", () => {
  describe("render", () => {
    it("should include NavLinks for all 5 sections", () => {
      const wrapper = shallow(<NavigationBar />);

      const navLinks = wrapper.find("NavLink");

      expect(navLinks).toHaveLength(3);
      expect(navLinks.at(0).prop("to")).toBe("/home");
      expect(navLinks.at(1).prop("to")).toBe("/about");
      expect(navLinks.at(2).prop("to")).toBe("/other");
    });
  });

  it("should match the snapshot", () => {
    const component = renderer.create(
      <Router>
        <NavigationBar />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
