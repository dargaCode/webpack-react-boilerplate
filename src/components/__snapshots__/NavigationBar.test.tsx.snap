// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar/> should match the snapshot 1`] = `
<header>
  <nav>
    <ul>
      <li>
        <a
          aria-current={null}
          href="/home"
          onClick={[Function]}
        >
          home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/about"
          onClick={[Function]}
        >
          about
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/other"
          onClick={[Function]}
        >
          other?
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
